File: EarningCalc_MW/Python/earningcalc.py
import datetime
import os

hourly_wage = 18
overtime_wage = 27

txt_file = "hoursTotal.txt"

def get_day():
    x = datetime.datetime.now()
    current_day = ((x.strftime("%A")) + ", " + (x.strftime("%B ")) + (x.strftime("%d")))
    print(current_day)

get_day()

today_hours = float(input("How many hours did you work today?: "))

def get_weekly_hours():
    with open(txt_file, "r") as file:
        week_hours_str = file.read().strip()
        week_hours_float = float(week_hours_str)
    return week_hours_float

week_hours = get_weekly_hours()
 
def earning_calc():
    total_hours = today_hours + week_hours
    if total_hours >= 40:
        normal_pay = 40 * hourly_wage
        overtime_hours = total_hours - 40
        overtime_pay = overtime_hours * overtime_wage
        gross_pay = normal_pay + overtime_pay
    if total_hours <= 40:
        gross_pay = total_hours * hourly_wage
        gp_str = str(gross_pay)
    print("The aproximate grossing eanred this week is:", gp_str)
    total_hours_str = str(total_hours)
    return total_hours_str

update_wk_hrs = earning_calc()


def write_to_dayshours():
    with open(txt_file, "w") as file:
        file.write(update_wk_hrs)
        file.close()

write_to_dayshours()

#print("Current working directory:", os.getcwd())

File: EarningCalc_MW/CPP/earningCalc.cpp
#include <iostream>
#include <fstream>
#include <ctime>
#include <string>
using namespace std;

void getDay() {
    time_t curr_time = time(nullptr);
    tm* dateTime = localtime(&curr_time);

    const char* dayNames[] = {"Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"};

    const char* monthNames[] = {"January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"};

    int weekday = dateTime->tm_wday;
    int month = dateTime->tm_mon;
    int monthDay = dateTime->tm_mday;

    cout << dayNames[weekday] << ", " << monthNames[month] << " " << monthDay << "\n";
}

float hoursInput() {
    float hours;
    cout << "How many hours did you work today?\n";
    cin >> hours;
    return hours;
}

float getWeeklyHours() {
    string weeklyHours_Str;
    ifstream fileReader("hours.txt");
    
    if (fileReader.is_open()) {
        getline(fileReader, weeklyHours_Str);
        fileReader.close();

        float weeklyHours_Flt = stof(weeklyHours_Str);
        return weeklyHours_Flt;
    } else {
        cout << "File not found, unable to read\n";
        return -1.0;
    }
}

string earningCalc() {
    const int hourlyWage = 18;
    const int overtimeWage = 27;
    float dayHours = hoursInput();
    float weekHours = getWeeklyHours();
    float hoursToDate = dayHours + weekHours;
    float grossPay;

    if (hoursToDate >= 40) {
        float normalMoney = 40 * hourlyWage;
        float overtimeHours = hoursToDate - 40;
        float overtimeMoney = overtimeHours * overtimeWage;
        grossPay = normalMoney + overtimeMoney;
    } else {
        grossPay = hoursToDate * hourlyWage;
    }

    cout << "Aproximate hours worked this week: " << hoursToDate << "\n";
    cout << "Aproximate grossing money earned this week: " << grossPay <<"\n";
    string hoursToDate_Str = to_string(hoursToDate);
    return hoursToDate_Str;
}

int main() {
    getDay();
    string totalHours_Str = earningCalc();

    ofstream fileWriter("hours.txt");
    
    if (fileWriter.is_open()) {
        fileWriter << totalHours_Str;
        fileWriter.close();
    } else {
        cout << "File not found, unable to write to file\n";
    }

}